src/worker/utils/chem.ts

```diff
@@ -136,6 +136,99 @@ export const getMatchingSubstructure = ({ structure, substructure }: { structure
   return { matchingAtoms: atoms, matchingBonds: bonds };
 };
 
+export const isValidMolBlock = (mdl: string) => {
+  if (!mdl.includes('M  END')) return false;
+  const mol = get_molecule(mdl, globalThis.workerRDKit);
+  if (!mol) return false;
+  try {
+    return mol.is_valid();
+  } finally {
+    release_molecule(mol);
+  }
+};
+
+/**
+ * Convert molecule structure to given notation (smiles, smarts, molblock, ...)
+ * TODO some notations like inchi don't work with qmol
+ */
+export const convertMolNotation = ({
+  moleculeString,
+  targetNotation,
+  sourceNotation,
+  useQMol = false,
+}: {
+  moleculeString: string;
+  targetNotation: MolNotation;
+  sourceNotation?: SourceMolNotation;
+  useQMol?: boolean;
+}): string | null => {
+  if (sourceNotation != null) {
+    if (sourceNotation === targetNotation) throw new Error('Source and target notations must differ');
+    if (!_validateSource(moleculeString, sourceNotation)) throw new Error('Molecule string not valid');
```

Comment:
We should create custom error for the package, or at least append [@iktos-oss/rdkit-provider] to the message, so that it easily identified and in the case of the custom error, we can easily catch/handle it